src/kontrol/__main__.py

```diff
@@ -0,0 +1,683 @@
+from __future__ import annotations
+
+import argparse
+import json
+import logging
+import re
+import sys
+from argparse import ArgumentParser
+from typing import TYPE_CHECKING
+
+from kevm_pyk.cli import KEVMCLIArgs, node_id_like
+from kevm_pyk.dist import DistTarget
+from kevm_pyk.utils import arg_pair_of
+from pyk.cli.utils import file_path
+from pyk.proof.tui import APRProofViewer
+
+from .foundry import (
+    Foundry,
+    foundry_get_model,
+    foundry_kompile,
+    foundry_list,
+    foundry_merge_nodes,
+    foundry_node_printer,
+    foundry_prove,
+    foundry_remove_node,
+    foundry_section_edge,
+    foundry_show,
+    foundry_simplify_node,
+    foundry_step_node,
+    foundry_to_dot,
+)
+from .solc_to_k import solc_compile, solc_to_k
+
+if TYPE_CHECKING:
+    from argparse import Namespace
+    from collections.abc import Callable, Iterable
+    from pathlib import Path
+    from typing import Any, Final, TypeVar
+
+    from pyk.cterm import CTerm
+    from pyk.kcfg.kcfg import NodeIdLike
+    from pyk.kcfg.tui import KCFGElem
+    from pyk.utils import BugReport
+
+    T = TypeVar('T')
+
+_LOGGER: Final = logging.getLogger(__name__)
+_LOG_FORMAT: Final = '%(levelname)s %(asctime)s %(name)s - %(message)s'
+
+
+def _ignore_arg(args: dict[str, Any], arg: str, cli_option: str) -> None:
+    if arg in args:
+        if args[arg] is not None:
+            _LOGGER.warning(f'Ignoring command-line option: {cli_option}')
+        args.pop(arg)
+
+
+def main() -> None:
+    sys.setrecursionlimit(15000000)
+    parser = _create_argument_parser()
+    args = parser.parse_args()
+    logging.basicConfig(level=_loglevel(args), format=_LOG_FORMAT)
+
+    executor_name = 'exec_' + args.command.lower().replace('-', '_')
+    if executor_name not in globals():
+        raise AssertionError(f'Unimplemented command: {args.command}')
+
+    execute = globals()[executor_name]
+    execute(**vars(args))
+
+
+# Command implementation
+
+
+def exec_version(**kwargs: Any) -> None:
+    raise NotImplementedError()
+
+
+def exec_compile(contract_file: Path, **kwargs: Any) -> None:
+    res = solc_compile(contract_file)
+    print(json.dumps(res))
+
+
+def exec_solc_to_k(
+    contract_file: Path,
+    contract_name: str,
+    main_module: str | None,
+    requires: list[str],
+    imports: list[str],
+    target: DistTarget | None = None,
+    **kwargs: Any,
+) -> None:
+    if target is None:
+        target = DistTarget.HASKELL
+
+    k_text = solc_to_k(
+        definition_dir=target.get(),
+        contract_file=contract_file,
+        contract_name=contract_name,
+        main_module=main_module,
+        requires=requires,
+        imports=imports,
+    )
+    print(k_text)
+
+
+def exec_foundry_kompile(
+    foundry_root: Path,
+    includes: Iterable[str] = (),
+    regen: bool = False,
+    rekompile: bool = False,
+    requires: Iterable[str] = (),
+    imports: Iterable[str] = (),
```

Comment:
We should rename the commands to be:

- `foundry-kompile => build`
- All other command sremove the `foundry` prefix.

Reply:
Done.